clippy_lints/src/legacy_numeric_constants.rs

```diff
@@ -113,35 +113,18 @@ impl<'tcx> LateLintPass<'tcx> for LegacyNumericConstants {
             // since this would only require removing a `use` import (which is already linted).
             && !is_numeric_const_path_canonical(path, [*mod_name, *name])
         {
-            (
-                vec![(expr.span, format!("{mod_name}::{name}"))],
-                "usage of a legacy numeric constant",
-            )
+            (format!("{mod_name}::{name}"), "usage of a legacy numeric constant")
         // `<integer>::xxx_value` check
         } else if let ExprKind::Call(func, []) = &expr.kind
             && let ExprKind::Path(qpath) = &func.kind
             && let QPath::TypeRelative(ty, last_segment) = qpath
             && let Some(def_id) = cx.qpath_res(qpath, func.hir_id).opt_def_id()
             && is_integer_method(cx, def_id)
+            && let Some(mod_name) = snippet_opt(cx, ty.span)
```

Comment:
`snippet_opt` will be removed [soon](https://github.com/rust-lang/rust-clippy/pull/15762)

```suggestion
            && let Some(ty_name) = ty.span.get_source_text(cx)
```

Reply:
I suppose alternatively we could also use `snippet_with_context` (with the ident's `SyntaxContext`), then we could still emit a suggestion for the test case and suggest `<$a>::MAX`, but just bailing out on macros seems fine too since it's unknown what other types will be passed to the macro.